### PR TITLE
Remove composableDerivation, closes #18763

### LINKED ISSUE #18763: Get rid of composableDerivation
A few packages use `composableDerivation`, which is complicated to understand. Since all other thousand packages use `.override` mechanism, we should convert the rest of dozen to do the same.

This is part of the effort to standardize the codebase.

Refs https://github.com/NixOS/nixpkgs/issues/18762

Scope:

```
$ git grep composableDerivation| wc -l
38
```

### PR DESCRIPTION
**Pull request by FRidh**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by domenkozar** [comment]
Should we provide an explanation (with assert) if someone references it?

**Comment by xeji** [comment]
The helper functions  `defaultOverridableDelayableArgs` and `overridableDelayableArgs` are still referenced in `lib/tests/misc.nix` and `lib/default.nix`.

**Review by matthewbauer — APPROVED** [review]

**Comment by FRidh** [comment]
> The helper functions defaultOverridableDelayableArgs and overridableDelayableArgs are still referenced in lib/tests/misc.nix and lib/default.nix.

Thanks for checking. Fixed.

File: lib/default.nix [hunk 1]
```diff
@@ -123,14 +123,14 @@ let
       traceShowValMarked showVal traceCall traceCall2 traceCall3
       traceValIfNot runTests testAllTrue traceCallXml attrNamesToStr;
     inherit (misc) maybeEnv defaultMergeArg defaultMerge foldArgs
-      defaultOverridableDelayableArgs composedArgsAndFun
+      composedArgsAndFun
```
Comment (xeji) [inline lib/default.nix]: `composedArgsAndFun` is removed from `lib/deprecated.nix` below.

File: lib/deprecated.nix [hunk 2]
```diff
@@ -285,7 +217,7 @@ rec {
   # };
   # will result in
   # { mergeAttrsBy = [...]; buildInputs = [ a b c d ]; }
-  # is used by prepareDerivationArgs, defaultOverridableDelayableArgs and can be used when composing using
+  # is used by prepareDerivationArgs and can be used when composing using
```
Comment (xeji) [inline lib/deprecated.nix]: `prepareDerivationArgs` is removed as well

File: lib/default.nix [hunk 3]
```diff
@@ -123,14 +123,14 @@ let
...
       maybeAttrNullable maybeAttr ifEnable checkFlag getValue
       checkReqs uniqList uniqListExt condConcat lazyGenericClosure
       innerModifySumArgs modifySumArgs innerClosePropagation
       closePropagation mapAttrsFlatten nvs setAttr setAttrMerge
       mergeAttrsWithFunc mergeAttrsConcatenateValues
       mergeAttrsNoOverride mergeAttrByFunc mergeAttrsByFuncDefaults
       mergeAttrsByFuncDefaultsClean mergeAttrBy prepareDerivationArgs
-      nixType imap overridableDelayableArgs;
```
Comment (xeji) [inline lib/default.nix]: `prepareDerivationArgs`  (L133) is removed

**Comment by FRidh** [comment]
Pushed an update. `lib` and its tests now evaluate.

**Pull request merged**